code-challenges.js

```diff
@@ -28,3 +61,34 @@ var hand2 = [5, 5, 3, 3, 4]
 // Expected output: false
 var hand3 = [5, 5, 5, 5, 4]
 // Expected output: false
+
+const fullHouse = (array) => {
+    // create an array to hold unique values
+    let uniqueCards = []
+    //map through array to find unique ranks of cards
+    //i admit that I received help from Brian during our study session haha. but we all took the time to teach each other why this works!
+    array.map(value => {
+        //push rank of card into uniqueCards if it doesnt already exist in that array
+        if (!uniqueCards.includes(value)) {
+            uniqueCards.push(value)
+        }
+    })
+    //compare original array to unique cards to see if duplicates
+    //if there are more than 2 ranks in hand, not possible to have a full house
+    if (uniqueCards.length !== 2) {
+        return false
+    }
+    //we only need index of unique cards
+    //for each index of unique cards, compare that unique card to each value of original array to check for pairs
+    //if a pair exists, it's ok if the other 3 cards are not a pair bc they are all the same card, thus being a triple
+    else if (uniqueCards.map((value, index) => array.filter(value => value === uniqueCards[index]).length === 2).includes(true)) {
+        return true
+    }
+    else {
+        return false
+    }
+}
```

Comment:
your pseudocode was so great in this file. it really communicated your code well. keep that up!